### PR TITLE
RadioGroup: Do not raise 'W1002' warning when reset() is called (T823478)

### PR DESCRIPTION
**Pull request by San4es**

### CODE REVIEW

**Review by alexander-kotov-dx — DISMISSED** [review]

File: js/ui/collection/ui.collection_widget.edit.js [hunk 1]
```diff
@@ -221,11 +221,16 @@ const CollectionWidget = BaseCollectionWidget.inherit({
...
+    _nullValueSelectionSupported: function() {
+        return false;
+    },
+
     _initSelectionModule: function() {
         const that = this,
             itemsGetter = that._editStrategy.itemsGetter;
 
         this._selection = new Selection({
+            allowNullValueSelection: this._nullValueSelectionSupported(),
```
Comment (San4es) [inline js/ui/collection/ui.collection_widget.edit.js]:
```suggestion
            allowNullValue: this._nullValueSelectionSupported(),
```
?

File: js/ui/selection/selection.js [hunk 2]
```diff
@@ -22,6 +22,7 @@ module.exports = Class.inherit({
 
     _getDefaultOptions: function() {
         return {
+            allowNullValueSelection: false,
```
Comment (San4es) [inline js/ui/selection/selection.js]:
```suggestion
            allowNullValue: false,
```
?

File: js/ui/selection/selection.strategy.standard.js [hunk 3]
```diff
@@ -132,9 +132,17 @@ module.exports = SelectionStrategy.inherit({
     },
 
     _warnOnIncorrectKeys: function(keys) {
+        var allowNullValueSelection = this.options.allowNullValueSelection;
```
Comment (San4es) [inline js/ui/selection/selection.strategy.standard.js]:
```suggestion
        var allowNullValue = this.options.allowNullValue;
```
?

File: js/ui/selection/selection.strategy.standard.js [hunk 4]
```diff
@@ -132,9 +132,17 @@ module.exports = SelectionStrategy.inherit({
...
+
         for(var i = 0; i < keys.length; i++) {
-            if(!this.isItemKeySelected(keys[i])) {
-                errors.log("W1002", keys[i]);
+            var key = keys[i];
+
+            if(key === null && allowNullValueSelection) {
```
Comment (San4es) [inline js/ui/selection/selection.strategy.standard.js]:
```suggestion
            if(key === null && allowNullValue) {
```

File: js/ui/selection/selection.strategy.standard.js [hunk 5]
```diff
@@ -132,9 +132,17 @@ module.exports = SelectionStrategy.inherit({
...
+                continue;
```
Comment (San4es) [inline js/ui/selection/selection.strategy.standard.js]: Merge with the `if` condition below

File: testing/tests/DevExpress.ui.widgets.editors/radioGroup.tests.js [hunk 6]
```diff
@@ -203,6 +204,30 @@ module("hidden input", () => {
 });
 
 module("value", moduleConfig, () => {
+
+    test("should not throw an error when the value is 'null'", assert => {
+        const errorLogStub = sinon.stub(errors, "log");
+
+        createRadioGroup({
+            items: ["1", "2", "3"]
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/radioGroup.tests.js]:
```suggestion
            items: ["1", "2", "3"],
            value: null
```

File: testing/tests/DevExpress.ui.widgets.editors/radioGroup.tests.js [hunk 7]
```diff
@@ -203,6 +204,30 @@ module("hidden input", () => {
...
+
+        createRadioGroup({
+            items: ["1", "2", "3"],
+            value: "2"
+        }).dxRadioGroup("reset");
+
+        assert.ok(errorLogStub.notCalled, "error was not thrown");
+        errorLogStub.restore();
+    });
+
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/radioGroup.tests.js:260]: Add test cases for `value` changing and `W1002` raising.

File: testing/tests/DevExpress.ui.widgets.editors/radioGroup.tests.js [hunk 8]
```diff
@@ -203,6 +204,60 @@ module("hidden input", () => {
...
+    test("should throw the W1002 error when the value is unknown key", assert => {
+        const errorLogStub = sinon.stub(errors, "log");
+
+        createRadioGroup({
+            items: [{ value: "1" }, { value: "2" }],
+            valueExpr: "value",
+            value: "3"
+        });
+
+        assert.ok(errorLogStub.calledOnce, "error was not thrown");
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/radioGroup.tests.js]:
```suggestion
        assert.ok(errorLogStub.calledOnce, "error was thrown");
```

File: testing/tests/DevExpress.ui.widgets.editors/radioGroup.tests.js [hunk 9]
```diff
@@ -203,6 +204,60 @@ module("hidden input", () => {
...
+        const instance = getInstance(
+            createRadioGroup({
+                items: ["1", "2", "3"],
+                value: "2"
+            })
+        );
+
+        instance.option("value", null);
+
+        assert.notOk(errorLogStub.called, "error was not thrown");
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/radioGroup.tests.js]:
```suggestion
        assert.ok(errorLogStub.notCalled, "error was not thrown");
```

File: testing/tests/DevExpress.ui.widgets.editors/radioGroup.tests.js [hunk 10]
```diff
@@ -203,6 +204,60 @@ module("hidden input", () => {
...
+
+    test("should not throw the W1002 error when the value is 'null' (T823478)", assert => {
+        const errorLogStub = sinon.stub(errors, "log");
+
+        createRadioGroup({
+            items: ["1", "2", "3"],
+            value: null
+        });
+
+        assert.notOk(errorLogStub.called, "error was not thrown");
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/radioGroup.tests.js]:
```suggestion
        assert.ok(errorLogStub.notCalled, "error was not thrown");
```

**Review by jsdmitry — APPROVED** [review]

**Review by alexander-kotov-dx — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]

**Pull request merged**